sdk/storage/Azure.Storage.Common/src/Shared/ContentHasher.cs

```diff
@@ -89,17 +89,15 @@ private static void AssertResponseHashMatch(GetHashResult computedHash, Validati
 
             switch (algorithm.ResolveAuto())
             {
-                case ValidationAlgorithm.MD5:
-                    if (!Enumerable.SequenceEqual(
-                        computedHash.MD5,
+                case StorageChecksumAlgorithm.MD5:
+                    if (!computedHash.MD5.Span.SequenceEqual(
                         response.Headers.TryGetValue("Content-MD5", out byte[] md5) ? md5 : default))
                     {
                         throw Errors.HashMismatch("Content-MD5");
                     }
                     break;
```

Comment:
@KrzysztofCwalina I'm not the most familiar with Span<T> vs Memory<T> but it seems like all the utility methods are on Span<T> and developers are expected to convert a Memory<T> to a Span<T> to use them. Is this correct?